http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/KubernetesServiceInterp.scala

```diff
@@ -0,0 +1,412 @@
+import LeonardoService.includeDeletedKey
+import io.chrisdavenport.log4cats.StructuredLogger
+import org.broadinstitute.dsde.workbench.google2.KubernetesName
+import org.broadinstitute.dsde.workbench.google2.KubernetesSerializableName.NamespaceName
+import org.broadinstitute.dsde.workbench.leonardo.AppType.Galaxy
+import org.broadinstitute.dsde.workbench.leonardo.SamResource.{AppSamResource, PersistentDiskSamResource}
+import org.broadinstitute.dsde.workbench.leonardo.http.api.{CreateDiskRequest, DiskConfigRequest}
+import org.broadinstitute.dsde.workbench.leonardo.http.service.LeoKubernetesServiceInterp.LeoKubernetesConfig
+import org.broadinstitute.dsde.workbench.leonardo.model.PersistentDiskAction.AttachPersistentDisk
+import org.broadinstitute.dsde.workbench.leonardo.model.ProjectAction.CreatePersistentDisk
+import org.broadinstitute.dsde.workbench.leonardo.service.KubernetesService
+
+import scala.concurrent.ExecutionContext
+
+class LeoKubernetesServiceInterp[F[_]: Parallel](
+  protected val authProvider: LeoAuthProvider[F],
+  protected val serviceAccountProvider: ServiceAccountProvider[F],
+  protected val leoKubernetesConfig: LeoKubernetesConfig,
+  protected val publisherQueue: fs2.concurrent.Queue[F, LeoPubsubMessage]
+)(
+  implicit F: Async[F],
+  log: StructuredLogger[F],
+  dbReference: DbReference[F],
+  ec: ExecutionContext
+) extends KubernetesService[F] {
+
+  override def createApp(
+    userInfo: UserInfo,
+    googleProject: GoogleProject,
+    appName: AppName,
+    req: CreateAppRequest
+  )(
+    implicit as: ApplicativeAsk[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- as.ask
+      //TODO check SAM permissions
+      hasPermission <- F.pure(true)
+      _ <- if (hasPermission) F.unit else F.raiseError[Unit](AuthorizationError(Some(userInfo.userEmail)))
+
+      appOpt <- KubernetesServiceDbQueries.getActiveFullAppByName(googleProject, appName).transaction
+      _ <- appOpt.fold(F.unit)(c =>
+        F.raiseError[Unit](AppAlreadyExistsException(googleProject, appName, c.app.status, ctx.traceId))
+      )
+
+      samResourceId <- F.delay(AppSamResource(UUID.randomUUID().toString))
+//   TODO: notify SAM
+//      _ <- authProvider.notifyKubernetesClusterCreated(samResourceId)
+
+      saveCluster <- F.fromEither(getSavableCluster(userInfo, googleProject, ctx.now))
+      saveClusterResult <- KubernetesServiceDbQueries.saveOrGetForApp(saveCluster).transaction
+
+      clusterId = saveClusterResult.minimalCluster.id
+      saveNodepool <- F.fromEither(getUserNodepool(clusterId, userInfo, req, ctx.now))
+      nodepool <- nodepoolQuery.saveForCluster(saveNodepool).transaction
+
+      diskOpt <- req.diskConfig.traverse(diskReq =>
+        LeoKubernetesServiceInterp.processDiskConfigRequest(
+          diskReq,
+          googleProject,
+          false,
+          userInfo,
+          authProvider,
+          serviceAccountProvider,
+          leoKubernetesConfig.diskConfig
```

Comment:
Shouldn't `PersistentDiskConfig` value be coming from user's app creation request? `leoKubernetesConfig` seems to be defined at Leo boot time?

Reply:
deep down in the call tree, in the DiskServiceInterp, we consolidate the request and the config such that defaults are used if the req has None in various places